ledger-core/virtual/integration/method_test.go

```diff
@@ -348,867 +390,540 @@ func TestVirtual_CallMethodAfterPulseChange(t *testing.T) {
-			return true // resend
-		})
-		typedChecker.VCallResult.Set(func(res *payload.VCallResult) bool {
-			assert.Equal(t, callFlags, res.CallFlags)
+			runnerMock.AddExecutionClassify("Foo", test.flagsA, nil)
+			runnerMock.AddExecutionClassify("Bar", test.flagsB, nil)
 
-			switch res.Callee {
-			case objectAGlobal:
-				require.Equal(t, []byte("finish A.New"), res.ReturnArguments)
-				require.Equal(t, payload.CTConstructor, res.CallType)
-				require.Equal(t, server.GlobalCaller(), res.Caller)
-				require.Equal(t, outgoingA, res.CallOutgoing)
-			case objectBGlobal:
-				require.Equal(t, []byte("finish B.Foo"), res.ReturnArguments)
-				require.Equal(t, payload.CTMethod, res.CallType)
-				require.Equal(t, objectAGlobal, res.Caller)
-				require.Equal(t, server.GetPulse().PulseNumber, res.CallOutgoing.Pulse())
+			// checks
+			typedChecker := server.PublisherMock.SetTypedChecker(ctx, mc, server)
+			typedChecker.VCallRequest.Set(func(request *payload.VCallRequest) bool {
+				assertCalleeRequest(t, objectAGlobal, objectBGlobal, request, test.flagsA, server)
+				return true // resend
+			})
+
+			typedChecker.VCallResult.Set(func(res *payload.VCallResult) bool {
+				switch res.Callee {
+				case objectAGlobal:
+					require.Equal(t, []byte("finish A.Foo"), res.ReturnArguments)
+					assertCallerResponse(t, res, server, test.flagsA)
+				case objectBGlobal:
+					require.Equal(t, []byte("finish B.Bar"), res.ReturnArguments)
+					assertCalleeResponse(t, res, objectAGlobal, test.flagsA)
+				default:
+					t.Fatalf("wrong Callee")
+				}
+				// we should resend that message only if it's CallResult from B to A
+				return res.Caller == objectAGlobal
+			}).ExpectedCount(2)
 
-			default:
-				t.Fatalf("wrong Callee")
+			pl := payload.VCallRequest{
+				CallType:            payload.CTMethod,
+				CallFlags:           payload.BuildCallFlags(test.flagsA.Interference, test.flagsA.State),
+				Caller:              server.GlobalCaller(),
+				Callee:              objectAGlobal,
+				CallSiteDeclaration: class,
+				CallSiteMethod:      "Foo",
+				CallOutgoing:        server.RandomLocalWithPulse(),
+				Arguments:           insolar.MustSerialize([]interface{}{}),
 			}
-			// we should resend that message only if it's CallResult from B to A
-			return res.Caller == objectAGlobal
-		})
-	}
-
-	pl := payload.VCallRequest{
-		CallType:       payload.CTConstructor,
-		CallFlags:      callFlags,
-		Caller:         server.GlobalCaller(),
-		Callee:         classA,
-		CallSiteMethod: "New",
-		CallOutgoing:   outgoingA,
-	}
-	msg := server.WrapPayload(&pl).Finalize()
-	server.SendMessage(ctx, msg)
-
-	// wait for all calls and SMs
-	testutils.WaitSignalsTimed(t, 10*time.Second, executeDone)
-	testutils.WaitSignalsTimed(t, 10*time.Second, server.Journal.WaitAllAsyncCallsDone())
-
-	require.Equal(t, 1, typedChecker.VCallRequest.Count())
-	require.Equal(t, 2, typedChecker.VCallResult.Count())
```

Comment:
use this structure instead of it ExpectedCount and PulisherMock.WaitCount

Reply:
ok